streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -919,7 +920,8 @@ public void run() {
      * @throws IllegalArgumentException if {@code timeout} can't be represented as {@code long milliseconds}
      */
     public synchronized boolean close(final Duration timeout) throws IllegalArgumentException {
-        ApiUtils.validateMillisecondDuration(timeout, "timeout");
+        final String msgPrefix = prepareMillisCheckFailMsgPrefix(timeout, "KafkaStreams#close()", "timeout");
```

Comment:
So we need to pass in the method name? The StackTrace will contain the method name anyway, won't it?

Reply:
yes, stack-trace will have the call chain. I thought, this would make easier for developers to read and identify the issue. 

these are the following options I can think of
1. Keep both method and class name. ex: `call(timeout, "KafkaStreams#close()", "timeout")`
2. Keep only class name. ex: `call(timeout, "KafkaStreams", "timeout")`
3. Remove both class and method then change text of   `MILLISECOND_VALIDATION_FAIL_MSG_FRMT` . ex: `call(timeout, "timeout")`

Which you think is better for us ?

Reply:
Personally, I prefer (3) because the others provide redundant information.